theseus/utils/examples/pose_graph/pose_prior_error.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Tuple
+
+import torch
+
+import theseus as th
+
+
+class PosePirorError(th.CostFunction):
```

Comment:
typo: `PosePirorError` -> `PosePriorError`
We can also drop the error since it derives from cost and is implied. Eventually we will move this to embodied dir as well.

Reply:
Actually, is this needed? The functionality is similar to [variable_difference.py](https://github.com/facebookresearch/theseus/blob/main/theseus/embodied/misc/variable_difference.py).

Which maybe we should rename to Manifold difference if optim vars will be restricted to Manifold class later and add this to the related item in #100.

Reply:
It seems that the `jacobian` of `variable_difference` is incorrect. Maybe we need to fix this.

Reply:
Is it incorrect for some classes? What is the error?